server/src/infra/repositories/filesystem.provider.ts

```diff
@@ -148,6 +148,6 @@ export class FilesystemProvider implements IStorageRepository {
     watcher.on(StorageEventType.UNLINK, (path) => events.onUnlink?.(path));
     watcher.on(StorageEventType.ERROR, (error) => events.onError?.(error));
 
-    return () => watcher.close();
+    return async () => await watcher.close();
```

Comment:
You can actually keep this as-is or you might incur the wrath of @jrasm91 haha

Reply:
phew @mertalev , you really saved my bacon here ;)